lib/index.js

```diff
@@ -469,6 +471,55 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
+   *
+   * @returns {Promise.<{ hits, total }>}
+   */
+  async search (searchBody, { from, size } = {}) {
+    debug('exists(searchBody = %j)', searchBody);
+
+    const forbiddenProperties = [
+      'multi_match',
+      ...Object.keys(storageMapping.users.properties)
+    ];
+
+    let isUsernameInQuery = false;
+    const sanitizedQuery = objectRecursiveMap(searchBody, (key, value) => {
+      if (key === 'username') {
+        isUsernameInQuery = true;
+        return { key: '_id' };
+      }
+      else if (value === 'username') {
+        isUsernameInQuery = true;
+        return { value: '_id' };
+      }
+
+      if (forbiddenProperties.some((property) => property === key || property === value)) {
+        throw new this.context.errors.BadRequestError(
+          `Forbidden keyword "${key}". Search query must only concern username property`);
```

Comment:
1. the request is well-formed: this should be a [ForbiddenError error](https://docs.kuzzle.io/core/2/framework/error-classes/forbidden-error/constructor)
2. the message could be better formatted
3. (nitpick / note) you don't have to wrap lines containing only a string

```suggestion
        throw new this.context.errors.ForbiddenError(
          `Forbidden field "${key}". Only the "username" field is searchable.`);
```